### PR TITLE
[lang] remove language add-on migration code

### PR DESCRIPTION
**Pull request by tamland**
It's been 3 version since we migrated. No need for this anymore

### CODE REVIEW

**Review by MartijnKaijser — APPROVED** [review]

**Comment by enen92** [comment]
Is this change for `strings.xml` or for the language folder layout (`resources/language/English/strings.po`)? or both?
Thanks

**Comment by MartijnKaijser** [comment]
It's the handling of the transfer from when we had language files shipped with Kodi itself towards the Kodi language addons to find the matching one and download it. 
This shouldn't affect strings for addons itself.

File: xbmc/LangInfo.cpp [hunk 1]
```diff
@@ -666,102 +666,23 @@ std::string CLangInfo::GetEnglishLanguageName(const std::string& locale /* = ""
...
-bool CLangInfo::SetLanguage(bool& fallback, const std::string &strLanguage /* = "" */, bool reloadServices /* = true */)
-{
-  fallback = false;
-
-  std::string language = strLanguage;
   if (language.empty())
-  {
     language = CServiceBroker::GetSettings().GetString(CSettings::SETTING_LOCALE_LANGUAGE);
 
-    if (language.empty())
```
Comment (Montellese) [inline xbmc/LangInfo.cpp]: Theoretically this could still happen.

**Review by Montellese — APPROVED** [review]
Looks ok.

File: xbmc/LangInfo.cpp (hunk 1 above)
Comment (MartijnKaijser) [inline xbmc/LangInfo.cpp]: Should we still keep this for that theoretical case?

Comment (tamland) [inline xbmc/LangInfo.cpp]: If it's empty it will print the below warning and fallback to default as it should. No need to have a special case for empty

**Pull request merged**